app/controllers/comments_controller.rb

```diff
@@ -1,13 +1,14 @@
 # frozen_string_literal: true
 
 class CommentsController < ApplicationController
+  before_action :set_giveaway, except: :comment_params
```

Comment:
That's inaccurate action to exclude, as per my previous comment, you are suppose to DRY the repetitive code and use it only in the actions where was used, i.e. **:new** and **:create**

Reply:
@OrceMaceski edit and update don't work without set_giveaway. Although one would think that find_comment is enough for those. That's why set_giveaway is needed for all the actions, except for the strong params, I thought you were suggesting that.

Reply:
The point of this **before_action** is to DRY the code, in this particular case I saw repetition of the code only in **new** and **create** action (before **set_giveaway** was created).

Reply:
@OrceMaceski   When `before_action :set_giveaway, only: [:new, :create, :find_comment]` actions `edit `and `update` don't work.

Reply:
you can add them ( `only: %i[new create edit update]` ) if they don't work though I can't see why because `@giveaway = Giveaway.find(params[:giveaway_id])` is not defined in those actions beforehand, are you sure they worked before?

Reply:
I found out that it's not possible to call a private method through an action or filter. Therefore, I found that what you are suggesting works like this: 
Instead of :
`before_action :set_giveaway, only: [:new, :create, :find_comment]`
`private`
`  def find_comment
      @comment = Comment.find(params[:id])
  end`
It works like this:
`before_action :set_giveaway, only: [:new, :create]`
`private`
`  def find_comment
      set_giveaway
      @comment = Comment.find(params[:id])
  end`

Reply:
> you can add them ( `only: %i[new create edit update]` ) if they don't work though I can't see why because `@giveaway = Giveaway.find(params[:giveaway_id])` is not defined in those actions beforehand, are you sure they worked before?

Yes, they worked before.